File: backend/src/main/java/com/simontrick/MadPaaToppen/processes/recipes/DTOs/RecipeResponse.java
package com.simontrick.MadPaaToppen.processes.recipes.DTOs;

import com.simontrick.MadPaaToppen.entities.Recipe;
import com.simontrick.MadPaaToppen.entities.DTOs.Instruction;

import java.util.List;
import java.util.UUID;

public record RecipeResponse(
        UUID id,
        String title,
        String description,
        List<Instruction> instructions,
        Integer prepTime,
        Integer cookTime,
        Integer servings,
        Integer version,
        List<RecipeVersionResponse> versions,
        List<RecipeIngredientResponse> ingredients
) {
    public RecipeResponse(Recipe recipe, 
        List<RecipeIngredientResponse> ingredients,
        List<RecipeVersionResponse> versions) {
        this(
                recipe.getId(),
                recipe.getTitle(),
                recipe.getDescription(),
                recipe.getInstructions(),
                recipe.getPrepTime(),
                recipe.getCookTime(),
                recipe.getServings(),
                recipe.getVersion(),
                versions,
                ingredients
        );
    }
}

File: frontend/src/types/recipeResponse.ts
export interface Recipe {
  id: string;
  title: string;
  description: string | null;
  instructions: Instruction[] | null;
  prepTime: number | null;
  cookTime: number | null;
  servings: number;
}

export interface RecipeResponse {
  id: string;
  title: string;
  description: string | null;
  instructions: Instruction[] | null;
  prepTime: number | null;
  cookTime: number | null;
  servings: number;
  ingredients: RecipeIngredient[] | null
}

export interface Instruction {
  step: number;
  text: string;
}

export interface Ingredient {
  name: Recipe;
  category: string;
}

export interface Unit {
  name: string;
  abbr: string;
}

export interface RecipeIngredient {
  recipe: Recipe;
  ingredient: Ingredient;
  quantity: number | null;
  unit: Unit;
  note: string;
}